operator/webhook/introute/sync.py

```diff
@@ -168,14 +197,15 @@ def _get_java_jdk_options(parent) -> Optional[Mapping[str, str]]:
 
     if not tls_config:
         return None
-
-    tls_type = tls_config['type']
-    assert tls_type in ["jks", "pkcs12"], \
-        f"({tls_type}) is not a supported TLS type. Supported types: ('jks', 'pkcs12')"
-
-    truststore_password = "changeit" if tls_type == "jks" else ""
-    jdk_options = f"-Djavax.net.ssl.trustStore={str(Path(TRUSTSTORE_PATH, tls_config['key']))} -Djavax.net.ssl.trustStorePassword={truststore_password} -Djavax.net.ssl.trustStoreType={tls_type.upper()}"
-
+    if 'clientConfig' in tls_config:
+        tls_type = tls_config['clientConfig']['type']
+        assert tls_type in ["jks", "pkcs12"], \
+            f"({tls_type}) is not a supported TLS type. Supported types: ('jks', 'pkcs12')"
+
+        truststore_password = "changeit" if tls_type == "jks" else ""
+        jdk_options = f"-Djavax.net.ssl.trustStore={str(Path(TRUSTSTORE_PATH, tls_config['clientConfig']['key']))} -Djavax.net.ssl.trustStorePassword={truststore_password} -Djavax.net.ssl.trustStoreType={tls_type.upper()}"
+    if 'serverConfig' in tls_config:
+        jdk_options +=  f" -Djavax.net.ssl.keyStore={str(Path(KEYSTORE_PATH, tls_config['serverConfig']['key']))}"
```

Comment:
It might make sense to add more unit tests for these scenarios

Reply:
Yeah, `NameError` I believe. I'll fix it up and add some more tests.

Reply:
Looking into this a bit more and I'm wondering if we should be configuring these properties using SSL bundles.
https://docs.spring.io/spring-boot/reference/features/ssl.html#features.ssl
In addition to the truststore and keystore configuration in that link, we should be able to specify our server trust bundle with the `server.ssl.bundle` property.

A complete example would be:
```
server.ssl.bundle=test
spring.ssl.bundle.jks.test.keystore.location=classpath:application.p12
spring.ssl.bundle.jks.test.keystore.password=secret
spring.ssl.bundle.jks.test.keystore.type=PKCS12
```

Reply:
I'm resolving this thread because a lot has changed since this discussion. Setting the truststore through JVM system args for clients to use and then setting the Spring `server.ssl.*` http server properties for the keystore maintains backwards compatibility.